### PR TITLE
Handle WomCoproductValue in inputs

### PR DESCRIPTION
**Pull request by tmooney**
When testing out a change to a CWL workflow I ran into a RuntimeException:
```
java.lang.RuntimeException: Unhandled CwlExpressionCommandPart value 'WomCoproductValue(WomCoproductType(NonEmptyList(WomStringType, WomIntegerType)),WomString(42))' of type Coproduct[String, Int]
```
This simple CWL file will reproduce the error:
```cwl
#!/usr/bin/env cwl-runner

cwlVersion: v1.0
class: CommandLineTool

baseCommand: ["/bin/echo"]
arguments: ["I got this value:"]
stdout: response.txt
inputs:
    value:
        type:
            - string
            - int
            - "null"
        inputBinding:
            position: 1
outputs:
    response:
        type: stdout
```
with an appropriate input YAML, e.g.:
```yaml
value: 42
```

The first commit in this PR was enough to get past that error; however, there was another issue if one of the possible types was instead a `File`.  The file did not appear to be getting localized.  So, a second example:

```cwl
#!/usr/bin/env cwl-runner

cwlVersion: v1.0
class: CommandLineTool

baseCommand: ["/bin/echo"]
arguments: ["I got this value:"]
stdout: response.txt
inputs:
    value:
        type:
            - string
            - File
            - "null"
        secondaryFiles: [.fai, .ann, .index]
        inputBinding:
            position: 1
outputs:
    response:
        type: stdout
```

The second commit to this PR addresses that issue.  (Without this commit Cromwell did not crash when run locally without Docker, but did crash with an `IllegalArgumentException` from calling `subpath` with an invalid range [here](https://github.com/broadinstitute/cromwell/blob/384f0b8f22399342705dc43ab9dac2d6b16bbf3b/backend/src/main/scala/cromwell/backend/io/JobPathsWithDocker.scala#L57) with Docker.)

These changes were enough to get my workflow running, but I'm not sure if there are other changes that should be made to properly handle `WomCoproductValue`.  (For instance, I have not tried it in GCP yet.)

### CODE REVIEW

**Comment by tmooney** [comment]
A new centaur test has been added based on an existing one (`cwl_prefix_for_array`) that was from a previous example CWL of mine 😄 .  Looks like whatever was causing that timeout didn't recur this go around.

**Comment by cjllanwarne** [comment]
I'm currently testing this in #5153 - assuming that all passes I'll merge this - thanks!

File: centaur/src/main/resources/standardTestCases/cwl_input_typearray/input_typearray.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+value:
+  class: File
+  path: "centaur/src/main/resources/standardTestCases/cwl_input_typearray/input.txt"
```
Comment (cjllanwarne) [inline centaur/src/main/resources/standardTestCases/cwl_input_typearray/input_typearray.yml]: I think to test this properly we would also want another input that looks the same in the CWL but is input more like:
```
value:
  class: String
  path: "centaur/src/main/resources/standardTestCases/doesnt_exist.txt"
```

In other words - if we provide the `String` version of the input, Cromwell should not try to localize anything and things should still work just fine.

Comment (tmooney) [reply centaur/src/main/resources/standardTestCases/cwl_input_typearray/input_typearray.yml]: Sounds reasonable--added a second input and output to the test!

**Pull request merged**